src/traces/barpolar/hover.js

```diff
@@ -0,0 +1,78 @@
+/**
+* Copyright 2012-2018, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var barHover = require('../bar/hover');
+var makeHoverPointText = require('../scatterpolar/hover').makeHoverPointText;
+var Color = require('../../components/color')
+var Fx = require('../../components/fx');
+var Lib = require('../../lib');
+var isPtInsidePolygon = require('../../plots/polar/helpers').isPtInsidePolygon;
+
+module.exports = function hoverPoints(pointData, xval, yval, hovermode) {
+    var cd = pointData.cd;
+    var trace = cd[0].trace;
+    var t = cd[0].t;
+
+    var subplot = pointData.subplot;
+    var radialAxis = subplot.radialAxis;
+    var angularAxis = subplot.angularAxis;
+    var xa = subplot.xaxis;
+    var ya = subplot.yaxis;
+    var inboxFn = subplot.vangles ? isPtInsidePolygon : Lib.isPtInsideSector;
+
+    if(radialAxis.range[0] > radialAxis.range[1]) {
+        xval = -xval;
+        yval = -yval;
+    }
+
+    var rVal = Math.abs(radialAxis.g2p(Math.sqrt(xval * xval + yval * yval)));
+    var thetaVal = Math.atan2(yval, xval);
+
+    // TODO add padding around sector to show labels,
+    // when hovering "close to" them
+    var distFn = function(di) {
+        var rBnds = [di.s0, di.s1].map(radialAxis.c2p);
+        var thetaBnds = [di.p0, di.p1].map(angularAxis.c2g).map(Lib.rad2deg);
+        return inboxFn(rVal, thetaVal, rBnds, thetaBnds, subplot.vangles) ?
+            1 :
+            Infinity;
+    };
+
+    Fx.getClosest(cd, distFn, pointData);
+
+    // skip the rest (for this trace) if we didn't find a close point
+    if(pointData.index === false) return;
+
+    var index = pointData.index;
+    var cdi = cd[index];
+    var rg = radialAxis.c2g(cdi.s1);
+    // TODO include offset here?
+    var thetag = angularAxis.c2g(cdi.p);
+    var xp = xa.c2p(rg * Math.cos(thetag));
+    var yp = ya.c2p(rg * Math.sin(thetag));
+
+    // TODO use 'extents' like in Bar.hover?
+    pointData.x0 = pointData.x1 = xp;
+    pointData.y0 = pointData.y1 = yp;
```

Comment:
This is good, but I wonder if we can flip the hover label so it's outside the bar pointing toward it? For the most part this means to the left if the bar is on the left half of the plot, unless the bar is inverted... so here ideally the label would be on the left for both purple bars:
![barpolar-hover](https://user-images.githubusercontent.com/2678795/44890384-08fbc100-aca8-11e8-8d7e-7ed7f48ad0b4.gif)

Reply:
Good idea. Is this what you had in mind?

![peek 2018-08-31 14-46](https://user-images.githubusercontent.com/6675409/44930555-b44e5980-ad2c-11e8-8544-cf933010a718.gif)


-> https://github.com/plotly/plotly.js/pull/2954/commits/e6c4dadecc2ce0a2d79d1f9c86242aa21a7acd75